src/includes/proc-service.h

```diff
@@ -14,6 +18,41 @@
 
 class process_service;
 
+#if SUPPORT_CAPABILITIES
+// A thin wrapper around the structure to avoid having to worry about freeing
+// it in various places (move semantics and destructor will take care of it)
+struct cap_iab_wrapper {
+    cap_iab_wrapper() {}
+    cap_iab_wrapper(std::string const &str) noexcept {
+        if (str.empty()) return;
+        iab = cap_iab_from_text(str.c_str());
```

Comment:
Add a comment that makes it clear that `iab` may still be null after this (if the string is invalid or if memory allocation fails within the `cap_iab_from_text` function).